language/diem-framework/releases/src/lib.rs

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{bail, Result};
+use diem_types::transaction::ScriptFunction;
+use once_cell::sync::Lazy;
+use std::{convert::TryFrom, fs, path::PathBuf};
+use vm::file_format::CompiledModule;
+
+pub mod legacy;
+
+#[cfg(test)]
+mod tests;
+
+fn crate_root_path() -> PathBuf {
+    PathBuf::from(env!("CARGO_MANIFEST_DIR"))
+}
+
+fn version_path(ver: &str) -> PathBuf {
+    let mut path = crate_root_path();
+    path.push("versions");
+    path.push(ver);
+    path
+}
+
+pub fn list_all_versions() -> Result<Vec<String>> {
+    let mut versions = vec![];
+
+    for entry in fs::read_dir(&crate_root_path().join("versions"))? {
+        let entry = entry?;
+
+        if entry.file_type()?.is_dir() {
+            versions.push(entry.file_name().to_string_lossy().to_string());
+        }
+    }
+
+    Ok(versions)
+}
+
+pub fn load_modules(ver: &str) -> Result<Vec<Vec<u8>>> {
```

Comment:
QQ: Should the api take a string as input or a `major.minor.fix` as input? I guess either way is fine but curious to see which one would you prefer.

Reply:
I think in the longer term we'll definitely want a more refined version number, but "major.minor.fix" alone may not be enough for this interface. I feel like we may need to also support "unofficial" releases that don't really get deployed -- it could be a release someone generated locally for testing purposes, or a snapshot we want to keep for development purposes. Due to this reason I'd prefer string for now, and plan to narrow it down once we know understand the problem better.